doc/user/build-install.xml

```diff
@@ -252,195 +277,88 @@ Python 3.7.1
 
-      <screen>
-# <userinput>python setup.py install --version-lib</userinput>
-      </screen>
-
       <para>
-
-      This will install the &SCons; build engine
-      in the
-      <filename>/usr/lib/scons-&buildversion;</filename>
-      or
-      <filename>C:\Python27\scons-&buildversion;</filename>
-      directory, for example.
-
+      Another approach to multiple versions is to create
+      &Python; virtualenvs, and install different &SCons; versions in each.
+      A Python <firstterm>virtual environment</firstterm>
+      is a directory with an isolated set of Python packages,
+      where packages you install/upgrade/remove inside the
+      environment do not affect anything outside it,
+      and those you install/upgrade/remove outside of it
+      do not affect anything inside it.
+      In other words, anything you do with <command>pip</command>
+      in the environment stays in that environment.
+      The &Python; standard library provides a module called
+      <systemitem>venv</systemitem> for creating these
+      (<ulink url="https://docs.python.org/e/library/venv.html"/>),
+      although there are also other tools which provide more precise
+      control of the setup.
       </para>
 
       <para>
-
-      If you use the <option>--version-lib</option> option
-      the first time you install &SCons;,
-      you do not need to specify it each time you install
-      a new version.
-      The &SCons; <filename>setup.py</filename> script
-      will detect the version-specific directory name(s)
-      and assume you want to install all versions
-      in version-specific directories.
-      You can override that assumption in the future
-      by explicitly specifying the <option>--standalone-lib</option> option.
-
+      Using a virtualenv can be useful even for a single version of
+      &SCons;, to gain the advantages of having an isolated environment.
+      It also gets around the problem of not having administrative
+      privileges on a particular system to install a distribution
+      package or use <command>pip</command> to install to a
+      system location, as the virtualenv is completely under your control.
       </para>
 
-    </section>
-
-    <section>
-    <title>Installing &SCons; in Other Locations</title>
-
       <para>
-
-      You can install &SCons; in locations other than
-      the default by specifying the <option>--prefix=</option> option:
-
+      The following outline shows how this could be set up
+      on a Linux/POSIX system (the syntax will be a bit different
+      on Windows):
       </para>
 
       <screen>
-# <userinput>python setup.py install --prefix=/opt/scons</userinput>
+$ <emphasis>create virtualenv named scons3</emphasis>
+$ <emphasis>create virtualenv named scons4</emphasis>
+$ <userinput>source scons3/bin/activate</userinput>
+$ <userinput>pip install scons==3.1.2</userinput>
+$ <userinput>deactivate</userinput>
+$ <userinput>source scons4/bin/activate</userinput>
```

Comment:
can't you get away with just `scons3/bin/pip install scons=3.1.2` and skip activating and deactivating?

Reply:
There are people who claim they've never activated a virtualenv in their lives, and that it's completely unnecessary, so mostly yes.  I didn't write about that because I always activate (seems so much simpler) and I don't want to fill this very-early section with a billion choices (other places are simplified for that reason too).  I think you *do* need to reference the Python "in" the virtualenv (which could be a symlink or such) or the paths don't come out right, so the incantation would be a little different than you list.

Reply:
Nope. Anything you run in the bin of a virtualenv will just work..;)

```
% ~/.virtualenvs/huntpt/bin/pip install -U pip
Requirement already satisfied: pip in /Users/bdbaddog/.virtualenvs/huntpt/lib/python3.10/site-packages (22.3)
Collecting pip
  Downloading pip-22.3.1-py3-none-any.whl (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 5.4 MB/s eta 0:00:00
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 22.3
    Uninstalling pip-22.3:
      Successfully uninstalled pip-22.3
Successfully installed pip-22.3.1

```

It is simpler when you're bouncing back and forth between several virtualenvs.

Of course your example is correct, just thought this might simpler (though you don't get scons in your path, and explaining that's probably not worthwhile here in this doc.)